### PR TITLE
Redesign mobile table

### LINKED ISSUE #78: Implement new table designs for mobile
Implement mobile designs for the home page and make the app mobile friendly

image

image (1)

### PR DESCRIPTION
**Pull request by SimplyBallistic**
Closes #78
http://aqua-table.surge.sh

~New hook: useIsMobile: detects if user is on mobile (xs and below)~ Removed
Old: 
image
New:
image

Possible Improvement:
~There are extra scrollbars in the mobile version, when the mobile table should just be allowed to take all the height it needs. Could be fixed in this PR~ Fixed

### CODE REVIEW

File: packages/website/src/routes/Homepage/ReefTable/tableHead.tsx [hunk 1]
```diff
@@ -79,7 +98,7 @@ interface EnhancedTableProps {
     property: OrderKeys
   ) => void;
   order: Order;
-  orderBy: OrderKeys;
+  orderBy: string;
```
Comment (SimplyBallistic) [inline packages/website/src/routes/Homepage/ReefTable/tableHead.tsx]:
```suggestion
  orderBy: OrderKeys;
```

**Comment by ericboucher** [comment]
@SimplyBallistic I updated the deisgns a tiny bit after discussing with the designer, it might require a few tweaks to clean up.

Also, you should use the new util:
`const { name, region: regionName } = getReefNameAndRegion(reef);`

**Comment by ericboucher** [comment]
Eg. the spacing at the bottom of each row could probably be reduced @SimplyBallistic

File: packages/website/src/routes/Homepage/ReefTable/tableHead.tsx [hunk 2]
```diff
@@ -38,25 +38,21 @@ const EnhancedTableHead = (props: EnhancedTableProps) => {
   const headCells: HeadCell[] = [
     {
       id: "locationName",
-      numeric: false,
-      label: "SITE",
+      label: "REEF",
```
Comment (ericboucher) [inline packages/website/src/routes/Homepage/ReefTable/tableHead.tsx:41]: Let's keep the label "SITE"

**Comment by SimplyBallistic** [comment]
@ericboucher Fixed everything you mentioned. Let me know what you think

**Comment by ericboucher** [comment]
Looks great, what do you think of using the new site-name design for the desktop version as well?

File: packages/website/src/routes/Homepage/ReefTable/body.tsx [hunk 3]
```diff
@@ -14,14 +14,88 @@ import { TableRow as Row } from "../../../store/Homepage/types";
...
+  reef: { locationName, region },
+  ...rest
+}: {
+  reef: Row;
+  className: string;
+}) => {
+  const isMobile = useIsMobile();
+  return (
+    <TableCell
+      style={isMobile ? { width: "35%", paddingRight: 0 } : undefined}
```
Comment (ericboucher) [inline packages/website/src/routes/Homepage/ReefTable/body.tsx:37]: Now that we are doing the same thing on both, you can use a theme breakpoint instead, no?

File: packages/website/src/routes/Homepage/ReefTable/index.tsx [hunk 4]
```diff
@@ -65,12 +69,17 @@ const ReefTable = ({ openDrawer, classes }: ReefTableProps) => {
...
         <TableContainer>
-          <Table stickyHeader>
-            <EnhancedTableHead
-              order={order}
-              orderBy={orderBy}
-              onRequestSort={handleRequestSort}
-            />
+          <Table
+            stickyHeader
+            style={isMobile ? { tableLayout: "fixed" } : undefined}
```
Comment (ericboucher) [inline packages/website/src/routes/Homepage/ReefTable/index.tsx:74]: similarly here if it's just a style change, maybe we could use classes and theme breakpoint?

File: packages/website/src/routes/Homepage/ReefTable/body.tsx [hunk 5]
```diff
@@ -14,14 +14,88 @@ import { TableRow as Row } from "../../../store/Homepage/types";
...
+      {...rest}
+    >
+      <Typography align="left" variant="h6" color="textSecondary">
+        {locationName}
+      </Typography>
+
+      {locationName !== region && region && (
+        <p style={{ color: "gray", margin: 0 }}>{region}</p>
```
Comment (ericboucher) [inline packages/website/src/routes/Homepage/ReefTable/body.tsx:45]: Should we use Typography here?

File: packages/website/src/routes/Homepage/ReefTable/SelectedReefCard/__snapshots__/index.test.tsx.snap [hunk 6]
```diff
@@ -12,7 +12,7 @@ exports[`renders as expected 1`] = `
         color="textSecondary"
         variant="h5"
       >
-        Selected Site
+        Featured Site
```
Comment (SimplyBallistic) [inline packages/website/src/routes/Homepage/ReefTable/SelectedReefCard/__snapshots__/index.test.tsx.snap:15]: Not sure what caused this. Lets see what the circleci build looks like.

File: packages/website/src/routes/Homepage/ReefTable/body.tsx [hunk 7]
```diff
@@ -14,14 +16,92 @@ import { TableRow as Row } from "../../../store/Homepage/types";
...
+}) => {
+  return (
+    <TableCell className={classes.cellTextAlign}>
+      <Typography
+        variant="h6"
+        style={{ color }}
+        className={classes.numberCellsTitle}
+      >
+        {formatNumber(value, decimalPlaces)}
+        &nbsp;
```
Comment (SimplyBallistic) [inline packages/website/src/routes/Homepage/ReefTable/body.tsx:71]: Is this supposed to be there?

File: packages/website/src/routes/Homepage/ReefTable/body.tsx [hunk 8]
```diff
@@ -14,14 +16,92 @@ import { TableRow as Row } from "../../../store/Homepage/types";
...
+  classes: ReefTableBodyProps["classes"];
+}) => {
+  return (
+    <TableCell className={classes.nameCells}>
+      <Typography align="left" variant="h6" color="textSecondary">
+        {locationName}
+      </Typography>
+
+      {locationName !== region && region && (
+        <Typography style={{ color: "gray" }} variant="subtitle1">
```
Comment (SimplyBallistic) [inline packages/website/src/routes/Homepage/ReefTable/body.tsx:42]: I couldn't find a `color` property that is gray. Know of one? Or do we need to set the style manually?

**Pull request merged**